src/main/java/org/springframework/data/elasticsearch/core/mapping/ElasticsearchPersistentEntity.java

```diff
@@ -171,9 +171,35 @@ default ElasticsearchPersistentProperty getRequiredSeqNoPrimaryTermProperty() {
 
 	/**
 	 * returns the default settings for an index.
-	 * 
+	 *
 	 * @return settings as {@link Document}
 	 * @since 4.1
 	 */
 	Document getDefaultSettings();
+
+	/**
+	 * @return the routing specification for the entity
+	 * @since 4.2
+	 */
+	@Nullable
+	String getRouting();
+
+	/**
+	 * @return {@literal true} if this Entity has a routing configuration.
+	 * @since 4.2
+	 */
+	default boolean hasRouting() {
+		return getRouting() != null;
+	}
+
+	/**
+	 * Resolves the routing for a bean in a SpEL context if {@link #hasRouting()} is true. The value of
+	 * {@link #getRouting()} is taken as a SpEL expression will be resolved to a String. The #bean is set as the context
+	 * root and can be referenced with '#root' or '#this'.
+	 * 
+	 * @param bean the bean to resolve the routing for
+	 * @return routing value, may be {@literal null}
+	 */
+	@Nullable
+	String resolveRouting(Object bean);
```

Comment:
The fact that routing can be SpEL should not be exposed. Evaluation should happen inside of the `PersistentEntity`.

Reply:
done